### ISSUE TITLE
Is our calculation of SCOP correct for hybrid heat pumps?

### ISSUE THREAD

**Issue by marliekeverweij** [quintel/etsource#2706]
[Here](https://github.com/quintel/etengine/blob/master/app/models/qernel/node_api/capacity_production.rb#L102-L119) you can see our calculation for the yearly average COP (SCOP):

![Screenshot 2022-06-29 at 14 54 17](https://user-images.githubusercontent.com/32833996/176441335-95828342-5fe1-4f6d-a360-7855477842d2.png)

This calculation works for all-electric heat pumps, because then you do:
heat_output_share (= `1`) / electricity_input_share (= `non_ambient_share`)

For hybrid heat pumps, the `non_ambient_share` consists of both electricity and gas/hydrogen, right @antw ? If that's the case, then the calculation should be adjusted I think.

I started looking into this calculation, because I figured that the SCOP of hybrid and gas hybrid heat pumps are different and expected them to be the same. For buildings we are adding a hydrogen HHP and then the SCOP is 4.0. The gas HHP has a SCOP of 5.5 in buildings.

I am not sure who to assign, but added @DorinevanderVlies because for now! Feel free to reassign

Notifying @quintel/core

**Comment by DorinevanderVlies** [comment]
> because I figured that the SCOP of hybrid and gas hybrid heat pumps are different and expected them to be the same

I agree, if the HP part of gas and hydrogen HHP have the same efficiency and cut-of COP I would indeed expect the SCOP of both to be the same.

@marliekeverweij do you know the definition of  SCOP of HHP's? I see two options:
1. SCOP of HHP is about the whole HHP. So all non ambient energy in / heat out. 
2. SCOP of HHP is about the HP part. So electricity in / heat out for HP part.

SCOP of option 1 will be lower since the efficiency of the gas part of the HHP is lower than efficiency of the HP/electric part.

Equations:
1. `heat_output_share  / (electricity_input_share + gas_input_share + hydrogen_input_share)`
2. `heat_output_share (when electricity is used) / electricity_input_share`

I am not sure if it is possible/easy to determine `heat_output_share (when electricity is used)`

**Comment by mabijkerk** [comment]
@Charlottevm and I also ran in some problems trying to trace back the COP of hybrid heat pumps.

To be clear, the COP of hybrid heat pumps is calculated using fever. In the snippet @marliekeverweij shared you can see the following condition:

```ruby
if fever&.efficiency_based_on && fever&.efficiency_balanced_with
            # When a Fever participant defines which carriers are used by a heat pump, use these
            # rather than guessing.
            heat_pump_coefficient_of_performance
```

The hybrid heat pumps meet this conditions, see for example the `buildings_space_heater_hybrid_heatpump_air_water_electricity`:

```
- fever.efficiency_balanced_with = ambient_heat
- fever.efficiency_based_on = electricity
```

Then `heat_pump_coefficient_of_performance` is defined further below in the file:
```ruby
      def heat_pump_coefficient_of_performance
        secondary_carriers = inputs.map { |input| input.carrier.key } -
          [fever.efficiency_based_on, fever.efficiency_balanced_with]

        primary_carrier = input(fever.efficiency_based_on)
        secondary_conversions = secondary_carriers.sum { |carrier| input(carrier).conversion }

        if primary_carrier.conversion.zero?
          1.0
        else
          (1 - secondary_conversions) / primary_carrier.conversion
        end
```

I would have to take some more time to find out what this piece of code does exactly. I do however think @DorinevanderVlies poses an interesting question: in our calculation of (S)COP, are we trying to separate the heat pump and gas part or not? In the first case I would refine the suggestion from @DorinevanderVlies:

Instead of `heat_output_share (when electricity is used) / electricity_input_share`, I would go for `share_of_electricity_and_ambient_in_heat_output_share / electricity_input_share`.

At a glance it **seems** as though this may be exactly what fever does in the snippet above.

**Comment by mabijkerk** [comment]
So, to follow up on myself with a blank scenario for 2050 in FR_france:

<img width="1501" alt="Screenshot 2023-02-22 at 15 23 11" src="https://user-images.githubusercontent.com/67338510/220649598-630b52b3-d3bb-4e44-b240-c5e6b390f565.png">

If we take the share in heat output of ambient heat and electricity (thanks @DorinevanderVlies for the excellent input), then divide that by the input share of electricity, we get:

```
(0.633 + 0.18) / 0.18 = 4.52
```

And ETEngine shows:
<img width="1445" alt="Screenshot 2023-02-22 at 15 24 47" src="https://user-images.githubusercontent.com/67338510/220650171-992e73c7-d5db-4e1e-ad6d-87b849645560.png">

**Comment by mabijkerk** [comment]
Three questions that are still open and that should be tackled:

- [ ] Is this way of calculating the COP documented anywhere? If not, where should we add it?
- [ ] The COP changes when a share > 0% is set for this hybrid heating technology. This is likely because this activates the seasonal component to the technology. Can we confirm that this is the case?
- [ ] The useable_heat output in the present year is > 100%, but for the future year in with 0% share it is reset to 100%. Can we explain why this is the case?

**Comment by kaskranenburgQ** [comment]
**Introduction**

For a blank nl2019 scenario, the share of hybrid network gas heat pumps is 0.2%. 
The SCOP in the front-end is 2.9 in a blank nl2019 scenario. 
This is the outcome of  the query: 
`V(households_space_heater_hybrid_heatpump_air_water_electricity,coefficient_of_performance)`
Which is ~ 2.90 in this scenario.
In this scenario the node households_space_heater_hybrid_heatpump_air_water_electricity has the following in-and outputs.

![Image](https://github.com/user-attachments/assets/907cb0df-18bc-4795-91ac-fcf385985070)

When we try to calculate the COP by hand as @mabijkerk did with the following method:

> If we take the share in heat output of ambient heat and electricity (thanks DorinevanderVlies for the excellent input), then divide that by the input share of electricity, we get:

`(0.35 + 0.20)/ 0.2 = 2.75`

Which is lower than the 2.90 we query. 


**Explanation**

The function heat_pump_coefficient_of_performance, which has been cited above here as well works as follows:

Primary Conversion (Electricity): 91.51 / 458.19 ≈ 0.200
Secondary Conversions (Network Gas only): 192.49 / 458.19 ≈ 0.420

COP = (1 - Secondary Conversions) / Primary Conversion
COP = (1 - 0.420) / 0.200
COP = 0.580 / 0.200 ≈ 2.90

This is why we see 2.90. 

This function does not take into account that the efficiency of the network gas unit of the hybrid heatpump is 1.07. 

If we were to account for this, the adjusted Secondary conversion would be: 192.49*1.07 / 458.19 ≈ 0.450

Leading to:
COP = (1 - 0.450) / 0.200
COP = 0.550 / 0.200 ≈ 2.75
Which is the COP we would expect according to the calculation method of @mabijkerk
